loadtest/user/userentity/user.go

```diff
@@ -96,6 +100,76 @@ func (ue *UserEntity) Connect() <-chan error {
 	return ue.wsErrorChan
 }
 
+// DownloadFile will download a url into a buffer
+func downloadFile(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("File not found: " + url)
+	}
+
+	defer resp.Body.Close()
+
+	return ioutil.ReadAll(resp.Body)
+}
+
+func extractLinks(data []byte) []string {
+	tokenizer := html.NewTokenizer(strings.NewReader(string(data)))
+	links := []string{}
+	for {
+		switch tokenizer.Next() {
+		case html.ErrorToken:
+			// End of the document, we're done
+			return links
+		case html.StartTagToken:
+			tag, _ := tokenizer.TagName()
+			nodeType := ""
+			nodeValue := ""
+			if string(tag) != "script" {
+				continue
+			}
+			for {
+				key, val, more := tokenizer.TagAttr()
+				if string(key) == "src" {
+					nodeValue = string(val)
+				}
+				if string(key) == "type" {
+					nodeType = string(val)
+				}
+				if !more {
+					break
+				}
+			}
+			if nodeType == "text/javascript" {
+				links = append(links, nodeValue)
+			}
+		}
+	}
+}
+
+// FetchStaticAssets parses index.html and fetches static assets mentioned in it
+func (ue *UserEntity) FetchStaticAssets() error {
+	indexHTML, err := downloadFile(ue.client.Url)
```

Comment:
I am wondering if we might need to pass more than this. Do the requests need to be authenticated in order to fetch the proper page/assets?

Reply:
@streamer45 - nope. non authenticated.
the stuff that comes after (avatars) needs authentication

Reply:
We might want to use the same behaviour when switching channels/teams so support for auth would be nice to have. I think we should have everything we need in the `UserEntity`.

Reply:
I'm going to use `colly` for html scraping, so I'm deleting my `downloadFile` function for now.
When we add asset fetching to authenticated parts - I'll handle it properly :)

Reply:
Can you make a follow-up Jira ticket?

Reply:
Done. https://mattermost.atlassian.net/browse/MM-23432